File: 11.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        l, r = 0, len(height) - 1
        maxWater = -1
        while l < r:
            prod = min(height[l], height[r]) * (r - l)
            maxWater = max(maxWater, prod)
            if height[l] > height[r]:
                r -= 1
            else:
                l += 1
        return maxWater

File: 11.cpp
#include <math.h>
#include <vector>
using namespace std;

class Solution {
public:
  int maxArea(vector<int> &height) {
    int l = 0, r = height.size() - 1, maxWater = 0;
    for (; l < r;) {
      int prod = min(height[l], height[r]) * (r - l);
      maxWater = max(maxWater, prod);
      height[l] > height[r] ? --r : ++l;
    }
    return maxWater;
  }
};